src/NodePowerGovernorAgent.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "NodePowerGovernorAgent.hpp"
+
+#include <cfloat>
+#include <cmath>
+#include <algorithm>
+
+#include "PowerGovernor.hpp"
+#include "PlatformIOProf.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/CircularBuffer.hpp"
+#include "geopm/Agg.hpp"
+#include "geopm/Helper.hpp"
+#include "config.h"
+
+namespace geopm
+{
+    NodePowerGovernorAgent::NodePowerGovernorAgent()
+        : NodePowerGovernorAgent(PlatformIOProf::platform_io())
+    {
+
+    }
+
+    NodePowerGovernorAgent::NodePowerGovernorAgent(PlatformIO &platform_io)
+        : m_platform_io(platform_io)
+        , m_level(-1)
+        , m_is_converged(false)
+        , m_is_sample_stable(false)
+        , m_do_send_sample(false)
+        // Placeholder for min/max signal equivalents to existing PowerGovernor
+        , M_MIN_POWER_SETTING(0)
+        , M_MAX_POWER_SETTING(std::numeric_limits<double>::max())
+        , M_POWER_TIME_WINDOW(0.013)
+        , m_pio_idx(M_PLAT_NUM_SIGNAL)
+        , m_pio_ctl_idx(M_PLAT_NUM_CONTROL)
+        , m_agg_func(M_NUM_SAMPLE)
+        , m_num_children(0)
+        , m_do_write_batch(false)
+        , m_last_power_budget(NAN)
+        , m_power_budget_changed(false)
+        , m_epoch_power_buf(geopm::make_unique<CircularBuffer<double> >(16)) // Magic number...
```

Comment:
Magic taken straight from the existing power governor agent.  If this is not needed for ongoing experiments we can remove it.